numba/targets/arraymath.py

```diff
@@ -1380,6 +1380,81 @@ def np_vander_seq_impl(x, N=None, increasing=False):
     elif isinstance(x, (types.Tuple, types.Sequence)):
         return np_vander_seq_impl
 
+#----------------------------------------------------------------------------
+# Mathematical functions
+
+@register_jitable
+def np_interp_impl_inner(x, xp, fp, out_dtype):
+    x_arr = np.asarray(x)
+    xp_arr = np.asarray(xp)
+    fp_arr = np.asarray(fp)
+
+    if len(xp_arr) == 0:
+        raise ValueError('array of sample points is empty')
+
+    if len(xp_arr) != len(fp_arr):
+        raise ValueError('fp and xp are not of the same length.')
+
+    if xp_arr.size > 1 and not np.all(np.diff(xp_arr) > 0):
+        raise ValueError('The x-coordinates of the data points must be increasing')
+        # this is a strict and literal interpretation of NumPy docs:
+        # https://github.com/numpy/numpy/blob/v1.15.0/numpy/lib/function_base.py#L1192
+        #   "The x-coordinates of the data points, must be increasing if argument
+        #   `period` is not specified."
+        # NumPy does not check for this or enforce it; whereas NumPy will return output,
+        # this implementation will raise.
+
+    if xp_arr.size == 1:
+        return np.full(x_arr.shape, fill_value=fp_arr[0], dtype=out_dtype)
+    else:
+        out = np.empty(x_arr.shape, dtype=out_dtype)
+
+        for i in range(x_arr.size):
+            if x_arr.flat[i] > xp_arr[-1]:
```

Comment:
should these two conditions include equality as well? If the interpolation point is on the first/last point there's no need to search?

Reply:
Yes - I took this out and can't remember why.  Perhaps it was when I was int he throes of trying to implement 'left' and 'right' before aborting missing.  I will take a look.

Reply:
Fixed